debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -153,6 +150,9 @@ public void execute(ChangeEventSourceContext context) {
                         currentRedoLogSequences = getCurrentRedoLogSequences();
                         Stopwatch stopwatch = Stopwatch.reusable();
                         while (context.isRunning()) {
+                            // Calculate time difference before each mining session to detect time zone offset changes (e.g. DST) on database server
+                            streamingMetrics.calculateTimeDifference(getSystime(jdbcConnection));
```

Comment:
Is this costly potentially? I.e. is it something we actually want to do upon every iteration?

Reply:
If we believe this is costly but necessary on each iteration, a compromise would be to add the `SYSTIMESTAMP` as a column in the LogMiner query against `V$LOGMNR_CONTENTS` and then use the first row in the result-set to resolve the time difference.  If we detect a change from the prior iteration, we then set this in the metrics.  This has the benefit that we remove the additional query but that we also minimize the AtomicReference calls unless necessary.

Reply:
This sounds pretty good to me.

Reply:
Btw. do we have to do the same thing in the XStream-based change event source?

Reply:
No we don't.  A majority of these "extended" metrics were introduced solely for the LogMiner implementation and we never adjusted the XStream code to support these.

Reply:
The response time of this calculation is mainly driven by the single round trip to the database server to get the timestamp. Overall the impact should be negligible if we consider what else is going on in this loop. Namely:

* A bunch of separate round trips to the DB to call procedures in `DBMS_LOGMNR`
* Get current SCN
* Update `LOG_MINING_FLUSH` table
* Fetch and process `V$LOGMNR_CONTENTS`

If requested I can do some benchmarks.

Reply:
> Overall the impact should be negligible if we consider what else is going on in this loop.

Ok, let's go with it then.

> If requested I can do some benchmarks.

Not needed for now, I trust your judgement on this. May come back on this in other contexts, though :)